providers/ProgressProvider.tsx

```diff
@@ -86,7 +86,7 @@ export default function ProgressProvider({
   return (
     <ProgressContext.Provider
       value={{
-        progress: accountProgress,
+        progress: accountProgress ?? 'CH1INT1',
```

Comment:
By initializing the progress variable to `CH1INT1` on line 10, you can probably just do `progress: accountProgress` here.

Reply:
I tried this line of code `progress: accountProgress`, but it causes the following error.

```
Type 'string | undefined' is not assignable to type 'string'.
Type 'undefined' is not assignable to type 'string'.
```

So I can either: 
1. change the `accountProgress` type to string or,
2. Use this (`progress: accountProgress ?? 'CH1INT1'`) maneuver

From these two, I chose the second option because it kept the code clean while being least intrusive.

Reply:
thats so odd, doing some of my own testing, even when setting the useState of account progress to be only a string and defaulted to 'CH1INT1' i still recieve an error on line 46 of `initLocal` with a type error stating that progress cannot be of type `string | undefined` even though it seems to be that progress has type string set everywhere

Reply:
@benalleng 

That's because the progress in the `initLocal` function is equal to the return value of `getProgressLocal` whose return value can be `<string | undefined>`. That's why the error is showing.

![image](https://user-images.githubusercontent.com/85434418/236485289-0be77889-d3c9-4945-a119-dff8e5ae5340.png)

It can be taken as one more reason to not change `accountProgress` type, because of the number of changes it requires to implement this PR's change.

Reply:
I guess my question is why should get ProgressLocal() allow for an undefined type? shouldn't it also default to 'CH1INT1'? 
I guess my thinking is that any instance of progress possibly being undefined would be better handled if it defaulted back to 'CH1INT1' as that is equivalent to base/no progress.

Reply:
This seems to be a better approach to me at first glance.

But I noticed that "undefined" is used in the `getProgressLocal` function when setting res fails for some reason.

```js
export default async function getProgressLocal() {
  try {
    const res = localStorage.getItem('SavingSatoshiProgress') ?? 'CH1INT1'

    return res
  } catch (errors) {
    console.error(errors)
    return undefined
  }
}
```

So what should be the approach you would suggest:

1. Should we replace the undefined with "CH1INT1" even when `getProgressLocal` fails for unspecified reasons?